src/models/serialportctrl.ts

```diff
@@ -0,0 +1,182 @@
+import * as vscode from 'vscode';
+import * as _ from 'lodash';
+import * as os from 'os';
+import {Board} from './Interfaces/Board';
+import {PortOption} from './Interfaces/PortOption';
+
+const serialport = require('../../vendor/node-usb-native').SerialPort;
+
+interface SerialPortInfo {
+  comName: string;
+  manufacturer: string;
+  vendorId: string;
+  productId: string;
+}
+
+export class SerialPortCtrl {
+  private static _port: any;
+
+  static getPlatform() {
+    return os.platform();
+  }
+
+  static getComList(): Promise<SerialPortInfo[]> {
+    return new Promise(
+      (
+        resolve: (value: SerialPortInfo[]) => void,
+        reject: (error: Error) => void
+      ) => {
+        // tslint:disable-next-line: no-any
+        serialport.list((err: any, ports: SerialPortInfo[]) => {
+          if (err) {
+            reject(err);
+          } else {
+            resolve(ports);
+          }
+        });
+      }
+    );
+  }
+
+  static async chooseCOM(board: Board | undefined): Promise<string> {
+    return new Promise(
+      async (
+        resolve: (value: string) => void,
+        reject: (reason: Error) => void
+      ) => {
+        let comList;
+        try{
+          comList = await SerialPortCtrl.getComList();
+        } catch (err) {
+          return reject(err);
+        }
+
+        const az3166 = board;

```

Comment:
Serial port is a general utility class, and should not contain any specific device information. Move it to high level code

Reply:
Moved to IoT Device Workbench.